platform/macos/src/MGLMapView.h

```diff
@@ -403,6 +404,19 @@ MGL_EXPORT IB_DESIGNABLE
 - (void)setVisibleCoordinateBounds:(MGLCoordinateBounds)bounds edgePadding:(NSEdgeInsets)insets animated:(BOOL)animated;
 
 /**
+ Changes the receiver’s viewport to fit the given shape, with the specified heading,
+ and optionally some additional padding on each side.
+
+ @param shape The shape that the viewport will show in its entirety.
+ @param heading The heading of the viewport, measured in degrees clockwise from true north.
+ @param insets The minimum padding (in screen points) that will be visible
+    around the given coordinate bounds.
+ @param animated Specify `YES` to animate the change by smoothly scrolling and
+    zooming or `NO` to immediately display the given bounds.
+ */
+- (void)setVisibleShape:(MGLShape *)shape heading:(double)heading edgePadding:(NSEdgeInsets)insets animated:(BOOL)animated;
```

Comment:
This method is reminiscent of `-showAnnotation:`, which also adds the passed-in annotation (which may be a shape) if it isn't already on the map. It might be nice to generalize that method with these additional parameters, but I'm not sure it's necessary to have one that only changes the camera, since the developer can readily compose `-cameraThatFitsShape:heading:edgePadding:` with `-setCamera:animated:`. The reason `-setVisibleCoordinateBounds:edgePadding:animated:` exists is that there's already a `visibleCoordinateBounds` property. But there isn't inherently a "visible shape" the way there's a visible coordinate bounds.

Reply:
@1ec5 I added this method to help with testing in the macOS demo app. and didn't expect it to be added to the mobile SDKs. Maybe there is a case to add this functionality to `-showAnnotation` but I'd prefer to scope this PR down to just the `-cameraTheFitsShape` changes.
//cc @d-prukop

Reply:
We try to keep the iOS and macOS SDKs as close as possible. In fact, there’s an entire [project](https://github.com/mapbox/mapbox-gl-native/projects/10) tracking that work.

Ideally we would find a way to make `-cameraThatFitsShape:heading:edgePadding:` compatible with one of the `-setCamera:…` methods, perhaps `-setCamera:withDuration:animationTimingFunction:edgePadding:completionHandler:`, such that we could add the same method to both SDKs publicly.